### ISSUE TITLE
Bear produces empty compile_commands.json

### ISSUE THREAD

**Issue by adorostkar** [rizsotto/Bear#266]
Hi, I installed the latest bear `version 2.4.2` in $HOME/.local and tried compiling a rather big project using this

    bear -vvvv -libear $HOME/.local/lib64/bear/libbear.so --append make -C sw/make all

but it produces an empty compile_command.json

I am on CentOs, I checked `estatus | grep 'SELinux status'` which says `disabled`

The output of the debug is quite long but it only prints the environment and does not go any further. run_build finishes with exit code -11

Any help is appreciated. Thanks.

**Comment by ashkan2200** [comment]
Seems to be related to make exit code but if I run the make separately it works.

**Comment by rizsotto** [comment]
Hey @ashkan2200 , thanks for your report.

Bear has a test suite, which runs on every commit. That has basic checks about the generated outputs, exit status, etc... In order to help you, I need to know how your use-case is different from those test-cases.

I noted that, you are running CentOS. (What version?) You have installed Bear from source. (Did you try out with some simple project?) What compiler your project is using? Is `make` just a wrapper around another build tool, or it does the compilation itself? Does the build process involves environment variable settings? Is it open-source?

**Comment by ashkan2200** [comment]
Sorry for late answer. As it turns out the linux is Red Hat. 

    NAME="Red Hat Enterprise Linux Server"
    VERSION="7.6 (Maipo)"

I have installed Bear from source. 
The compiler used is arm-wrs-linux-gnueabi-gcc
make is doing the compilation. 
I am not sure about the environment variable setting but since I do it directly with make I guess it doesn't do that.
Unfortunately the project is not open source.

Thank you for the help

**Comment by ashkan2200** [comment]
With a simple hello world I still get empty compile_commands.json with the same error code -11
running make alone works.

**Comment by rizsotto** [comment]
Thanks to come back.

Could you check if the compiler is a statically linked binary? (`file $(which arm-wrs-linux-gnueabi-gcc)` or `ldd $(which arm-wrs-linux-gnueabi-gcc)` will tell you that.)

**Comment by ashkan2200** [comment]
here is the output
ELF 64-bit LSB executable, dynamically linked

**Comment by rizsotto** [comment]
You've mentioned that you tried to run Bear against a simple project. And that one also had empty output. Could you attach the verbose output of that run?

**Comment by ashkan2200** [comment]
Here is the log. 
https://drive.google.com/open?id=1lME7_SXDnCNsNxjAJHXL2sEZIIWrwOb0

**Comment by rizsotto** [comment]
Thanks, it was helpful.

The shell will not replace the `$HOME` with it's value. So, Bear was received the string `-l $HOME/.local/...` which it was also not look for environment to substitute. And it was end up as `LD_PRELOAD=ibear`, which cause the execution of `make` to fail.

As a workaround try to pass the full path to `libear.so` or `-l $(echo $HOME)/.local/...`.

**Comment by ashkan2200** [comment]
This is my mistake, I put up the wrong log. let me try again. Sorry for inconvenience.

**Comment by ashkan2200** [comment]
https://drive.google.com/open?id=1_CQqjqjAzIuRqvQJ4vgJuOpIt1H1yqzS
Ok this should be correct now. Sorry again

**Comment by rizsotto** [comment]
:) You are passing `-libear` instead of `-l` or `--libear`.

**Comment by ashkan2200** [comment]
Ok again my fault. Next error is this. When I actually pass correct argument
ERROR: ld.so: object '/home/ealidor/.local/lib64/bear/libear.so' from LD_PRELOAD cannot be preloaded: ignored.

**Comment by rizsotto** [comment]
Thanks to come back. Will try to help you tomorrow, I need to sleep now. Until then...

- Please send me the command line you've used.
- Please send me the output of these commands:
    - `file $(which make)`
    - `file $(which which arm-wrs-linux-gnueabi-gcc)`
    - `file /home/ealidor/.local/lib64/bear/libear.so`
    - `nm -DC /home/ealidor/.local/lib64/bear/libear.so`
- Try to look into these issues for ideas what could have gone wrong... #130 , #214, #187

**Comment by ashkan2200** [comment]
Figured one issue from #78. The make was 32-bit while the lib was 64. I switched the make and now the simple example is working. 

For the big project I had to switch several tools that was used in the make. The way I did it (which may help others) is to do 

    make -n -C sw/make all 

This produces a dry run with all the tools that are used. I checked the tools that are in the makefile and replaced all the 32bit tools with 64bit and it worked. 

Thank you for helping.
